.github/workflows/java-cd.yml

```diff
@@ -158,3 +143,112 @@ jobs:
+              run: |
+                cd maven-files
+                for i in *.jar *.pom *.module; do md5sum $i | cut -d ' ' -f 1 > $i.md5; done
+                for i in *.jar *.pom *.module; do sha1sum $i | cut -d ' ' -f 1 > $i.sha1; done
+
+            - name: Move files to the correct directory tree
+              run: |
+                  mkdir -p build/io/valkey/valkey-glide/${{ env.RELEASE_VERSION }}
+                  cp -a maven-files/* build/io/valkey/valkey-glide/${{ env.RELEASE_VERSION }}
+                  rm -rf build/io/valkey/valkey-glide/${{ env.RELEASE_VERSION }}/META-INF
+                  cd build
+                  zip -r ../build io
+
+            - name: Upload artifacts to publish
+              uses: actions/upload-artifact@v4
+              with:
+                name: valkey-${{ env.RELEASE_VERSION }}
+                path: |
+                  build
+
+            - name: Publish to Maven Central
+              id: maven-deployment
+              run: |
+                  BEARER_TOKEN=$(echo "${{ secrets.CENTRAL_TOKEN_USERNAME }}:${{ secrets.CENTRAL_TOKEN_PASSWORD }}" | base64)
+                  export DEPLOYMENT_ID=`curl --request POST \
+                  --header "Authorization: Bearer $BEARER_TOKEN" \
+                  --form bundle=@build.zip \
+                  https://central.sonatype.com/api/v1/publisher/upload | tail -n 1`
+                  echo "DEPLOYMENT_ID=$DEPLOYMENT_ID" >> $GITHUB_ENV
+                  echo "BEARER_TOKEN=$BEARER_TOKEN" >> $GITHUB_ENV
+                  echo "DEPLOYMENT_ID=$DEPLOYMENT_ID" >> $GITHUB_OUTPUT
+                  echo $DEPLOYMENT_ID
+
+            - name: Check status of deployment
+              run: |
+                  sleep 20
+                  export DEPLOYMENT_STATUS=`curl --request POST \
+                  --header "Authorization: Bearer ${{ env.BEARER_TOKEN }}" \
+                  "https://central.sonatype.com/api/v1/publisher/status?id=${{ env.DEPLOYMENT_ID }}" \
+                  | jq '.deploymentState'`
+                  echo $DEPLOYMENT_STATUS
+                  if [[ $DEPLOYMENT_STATUS != ""\"VALIDATED"\"" || $DEPLOYMENT_STATUS -eq null ]]; then exit 1 ; fi
+
+            - uses: actions/checkout@v4
+              with:
+                submodules: recursive
+
+            - name: Set up JDK
+              uses: actions/setup-java@v4
+              with:
+                distribution: "temurin"
+                java-version: '11'
+
+            - name: Install protoc (protobuf)
+              uses: arduino/setup-protoc@v3
+              with:
+                version: "26.1"
+                repo-token: ${{ secrets.GITHUB_TOKEN }}
+
+            - name: Test deployment
+              working-directory: java
+              run: |
+                  export ORG_GRADLE_PROJECT_centralManualTestingAuthHeaderName="Authorization"
+                  export ORG_GRADLE_PROJECT_centralManualTestingAuthHeaderValue="Bearer ${{ env.BEARER_TOKEN }}"
```

Comment:
Are those gradle predefined variable names?

Reply:
Yeah it comes from https://central.sonatype.org/publish/publish-portal-api/#gradle